pkg/expapi/v1/types.go

```diff
@@ -350,3 +350,105 @@ type ThirdPartyResourceDataList struct {
 	// Items is a list of third party objects
 	Items []ThirdPartyResourceData `json:"items"`
 }
+
+// Job represents the configuration of a single job.
+type Job struct {
+	v1.TypeMeta   `json:",inline"`
+	v1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec is a structure defining the expected behavior of a job.
+	Spec JobSpec `json:"spec,omitempty"`
+
+	// Status is a structure describing current status of a job.
+	Status JobStatus `json:"status,omitempty"`
+}
+
+// JobList is a collection of jobs.
+type JobList struct {
+	v1.TypeMeta `json:",inline"`
+	v1.ListMeta `json:"metadata,omitempty"`
+
+	Items []Job `json:"items"`
+}
+
+// JobSpec describes how the job execution will look like.
+type JobSpec struct {
+
+	// Parallelism specifies the maximum desired number of pods the job should
+	// run at any given time. The actual number of pods running in steady state will
+	// be less than this number when ((.spec.completions - .status.successful) < .spec.parallelism),
+	// i.e. when the work left to do is less than max parallelism.
+	Parallelism *int `json:"parallelism,omitempty"`
+
+	// Completions specifies the desired number of successfully finished pods the
+	// job should be run with. Defaults to 1.
+	Completions *int `json:"completions,omitempty" descrption:"desired number of pods"`
+
+	// Selector is a label query over pods that should match the pod count.
+	Selector map[string]string `json:"selector"`
+
+	// Template is the object that describes the pod that will be created when
+	// executing a job.
+	Template *v1.PodTemplateSpec `json:"template"`
+}
+
+// JobStatus represents the current state of a Job.
+type JobStatus struct {
+
+	// Conditions represent the latest available observations of an object's current state.
+	Conditions []JobCondition `json:"conditions,omitempty"`
+
+	// StartTime represents time when the job was started. It is not guaranteed to
+	// be set in happens-before order across separate operations.
+	// Clients may not set this value. It is represented in RFC3339 form and is in UTC.
```

Comment:
No need to state that Clients may not set this value.  In general, clients may not set JobStatus on a POST/PUT to `/expapi/v1/jobs/myjob`.   To set this, one must POST to  `/expapi/v1/jobs/myjob/status`.  I haven't looked at that part of the implementation PR, but if you know it doesn't do that validation, them please add it.  If you were copying ReplicationController, then I think you'll have got it right.